ml-agents/mlagents/trainers/components/reward_signals/curiosity/signal.py

```diff
@@ -0,0 +1,168 @@
+import numpy as np
+from mlagents.trainers.components.reward_signals import RewardSignal
+from mlagents.trainers.components.reward_signals.curiosity.model import CuriosityModel
+from mlagents.trainers.policy import Policy
+
+
+class CuriosityRewardSignal(RewardSignal):
+    def __init__(self, policy: Policy, strength, gamma, encoding_size=128, num_epoch=3):
+        """
+        Creates the Curiosity reward generator
+        :param policy: The Learning Policy
+        :param encoding_size: The size of the Curiosity encoding
+        :param signal_strength: The scaling parameter for the reward. The scaled reward will be the unscaled
+        reward multiplied by the strength parameter
+        """
+        self.policy = policy
+        self.strength = strength
+        self.gamma = gamma
+        self.stat_name = "Policy/Curiosity Reward"
```

Comment:
Do stat_name or value_name get used anywhere? Don't see them referenced for this or any of the other RewardSignal implementations.

If they are going to be used, maybe it would be better to have them as properties in the base class; you could munge them from the `cls.__name__` by default or allow them to be overridden.

Reply:
Hey @chriselion, they are used - every RewardSignal needs both of these. They're used for stats reporting.

I've added them to the base class, though rather than using the class name I'd rather have the "human friendly" name reported here (e.g. "Curiosity Reward" instead of "CuriosityRewardSignal"). So for now I'll override them - but maybe there's a better way?

Reply:
OK, I see where it's used now.

I would probably do something like
``` python
class RewardSignal:
  def __init__(self):
    class_name = self.__class__.__name__
    short_name = class_name.replace("RewardSignal", "")
    self.stat_name = f"Policy/{short_name} Reward"
    self. value_name = f"Policy/{short_name} Value Estimate"
```
And then call the base constructor in each child class. That should set everything up (assuming you rename EntropySignal) with a reasonable default but still allow it to be overridden if desired.

Reply:
This code works a treat, thanks!